### PR TITLE
Update notary private key implementation

### PR DESCRIPTION
**Pull request by sergiupuhalschi-rdx**

### CODE REVIEW

File: src/core/types/keys/ed25519/public_key.rs [hunk 1]
```diff
@@ -70,18 +70,12 @@ pub fn new_ed25519_public_key_sample_other() -> Ed25519PublicKey {
 }
 
 #[uniffi::export]
-pub fn android_notary_key_get_public_key_from_private_key_bytes(
-    private_key_bytes: Entropy32Bytes,
+pub fn android_secret_key_get_public_key_from_private_key_bytes(
+    private_key_bytes: Exactly32Bytes,
 ) -> Result<Ed25519PublicKey> {
-    let mut private_key_bytes = private_key_bytes;
```
Comment (Sajjon) [inline src/core/types/keys/ed25519/public_key.rs:76]: can probably be written as:

```rust
{
    Ed25519PrivateKey::try_from(private_key_bytes.as_ref()).map(|k| k.public_key())
```

File: src/wrapped_radix_engine_toolkit/low_level/notary_signature_uniffi_fn.rs [hunk 2]
```diff
@@ -24,18 +24,28 @@ pub fn notary_signature_get_signature(
...
+
+#[uniffi::export]
+pub fn android_sign_hash_with_private_key_bytes(
+    private_key_bytes: Exactly32Bytes,
+    hash: &Hash,
+) -> Result<Signature> {
+    let ed25519_private_key =
+        Ed25519PrivateKey::try_from(private_key_bytes.as_ref())?;
+
+    let private_key = PrivateKey::from(ed25519_private_key);
```
Comment (Sajjon) [inline src/wrapped_radix_engine_toolkit/low_level/notary_signature_uniffi_fn.rs:47]: this step is not needed, you can use `sign` on `ed25519_private_key` (rename it -> `private_key`) directly. 

And you should return type to `Result<Ed25519Signature>` instead, and maybe change the whole func to:

```rust
{
    Ed25519PrivateKey::try_from(private_key_bytes.as_ref()).map(|pk| pk.sign(hash))
}
```

File: src/core/types/keys/ed25519/public_key.rs (hunk 1 above)
Comment (sergiupuhalschi-rdx) [reply src/core/types/keys/ed25519/public_key.rs:76]: Wow, nice one. Thanks!

**Review by CyonAlexRDX — APPROVED** [review]
LGTM, good job!

**Review by micbakos-rdx — APPROVED** [review]

**Pull request merged**